proposals/22-04-localize-command.md

```diff
@@ -0,0 +1,331 @@
+different layers of kustomization files from sharing the same copy of the remote files. Following the same logic,
+different potential `target` directories cannot share copies either.
+
+## Drawbacks
+
+Users whose layered kustomizations form a complex directory tree structure may have a hard time finding an
+appropriate `scope`. However, many kustomizations exist in repositories, allowing the user to easily choose the repo
+root as a valid `scope`. The warning messages that `kustomize localize` outputs for reference paths that extend
+beyond `scope` should also help.
+
+## Alternatives
+
+* Instead of downloading into `newDir`, `kustomize localize` could download all remote files into a directory specified
+  by some global environment variable (like in Golang), which would preclude deeply nested directories and allow
+  different kustomization files to share configurations. On top of that, if `kustomize build` had the added
+  functionality to check for previous downloads of remote references at said global location, `kustomize localize` would
+  not need to overwrite the remote references in `target` to the local downloads. As a result, `kustomize localize`
+  would need to neither write to `target` nor copy `target` into `newDir`. The user would not need to specify a `scope`
+  either. <br></br>
+
+  Despite its advantages, the alternative design violates the self-contained nature of each kustomize layer. Users would
+  be unable to upload a fully localized kustomization directory in version control. Furthermore, this alternative
+  complicates the existing kustomize workflow by requiring the setup of global environment variables.
+  <br></br>
+
+* The command could, instead of making a copy, modify `target` directly. However, users would not have an easy way to
+  undo the command, which is undesirable.
+  <br></br>
+
+* Instead of requiring the user to specify a second argument `scope`, the command could by definition limit its copying
+  to `target`. However, in the case of **Story 1**, the command would force the user to set `target` to `example` in
+  order to include `example/base` in the localization of `example/overlay`. The user would then have to create a
+  kustomization file at `example` that points to `example/overlay` under the `resources` field. The creation of the
+  kustomization file solely for this purpose is messy and more work for the user.
+
+## Rollout Plan
+
+This command will have at least alpha and GA releases. Depending on user feedback, we may add a beta.
+
+### Alpha
+
+This release will not support
+
+* load restrictions check
+* KRM functions
+* absolute paths
+* `target` kustomization with reference cycles
+* any verification in the form of `--no-verification` flag or automatically running `kustomize build` at the end of
+  localization
+
+The entire command will be new in the alpha release, and so will not require an alpha flag. The command will not be
```

Comment:
Nit! Since we don't have an alpha command group, let's make sure the help text indicates it (maybe even print a warning/feedback solicitation on execution?)

Reply:
Added